docs/docs/documentation/components/proxies.md

```diff
@@ -0,0 +1,473 @@
+    artifact_service:
+      type: "filesystem"
+      base_path: "/tmp/proxy-artifacts"
+    discovery_interval_seconds: 60
+    default_request_timeout_seconds: 300
+
+broker:
+  # Broker configuration inherited from environment or specified here
+```
+
+### Configuration Parameters
+
+- `namespace`: The topic prefix for A2A communication (for example, "myorg/production").
+- `proxied_agents`: A list of external agents to proxy. Each agent can have its own URL, authentication, and timeout settings (see Authentication Types below).
+- `artifact_service`: Configuration for storing artifacts (memory, filesystem, or GCS). This is shared across all proxied agents.
+- `discovery_interval_seconds`: How often to refresh agent cards from all external agents (default: 60).
+- `default_request_timeout_seconds`: Default timeout for requests to external agents. Individual agents can override this (default: 300).
+
+## Authentication Types
+
+The proxy supports three authentication schemes for connecting to downstream agents. Each agent in the `proxied_agents` list can use a different authentication type, allowing you to integrate agents with varying security requirements in a single proxy instance.
+
+### Static Bearer Token
+
+Use static bearer tokens for agents that require a fixed authentication token.
+
+```yaml
+proxied_agents:
+  - name: "secure-agent"
+    url: "https://api.example.com/agent"
+    authentication:
+      type: "static_bearer"
+      token: "${AGENT_BEARER_TOKEN}"  # Use environment variable
+```
+
+### Static API Key
+
+Use static API keys for agents that require API key authentication.
+
+```yaml
+proxied_agents:
+  - name: "api-key-agent"
+    url: "https://api.example.com/agent"
+    authentication:
+      type: "static_apikey"
+      token: "${AGENT_API_KEY}"
+```
+
+### OAuth 2.0 Client Credentials
+
+Use OAuth 2.0 client credentials flow for agents that require dynamic token acquisition. The proxy automatically handles token refresh and retry logic.
+
+```yaml
+proxied_agents:
+  - name: "oauth-agent"
+    url: "https://api.example.com/agent"
+    authentication:
+      type: "oauth2_client_credentials"
+      token_url: "https://auth.example.com/oauth/token"
+      client_id: "${OAUTH_CLIENT_ID}"
+      client_secret: "${OAUTH_CLIENT_SECRET}"
+      scope: "agent.read agent.write"  # Optional
+      token_cache_duration_seconds: 3300  # Optional, default: 3300 (55 minutes)
+```
+
+The proxy caches OAuth tokens and automatically refreshes them when they expire. If a request receives a 401 Unauthorized response, the proxy invalidates the cached token and retries the request once with a fresh token.
+
+:::note[Security Best Practice]
+Always use environment variables for sensitive credentials. Never commit tokens or secrets directly in configuration files.
+:::
+
+## Artifact Handling
+
+The proxy manages artifact flow in both directions to ensure seamless integration between the mesh and external agents.
```

Comment:
Minor comment but given that the docs refer to both Agent Mesh and event mesh, it might be best to use one or the other (whichever is appropriate for each instance) consistently to prevent confusion.